shell/shell.go

```diff
@@ -0,0 +1,104 @@
+package shell
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	"sync"
+
+	"github.com/google/shlex"
+)
+
+// Default Inteface to use if none provided.
+var Default = "bash"
+
+// Interface to the shell.
+type Interface interface {
+	// Name of the shell interface.
+	Name() string
+
+	// Put in a exec command line and get a Cmd.
+	Command(line string) (*exec.Cmd, error)
+}
+
+var (
+	shellsLock sync.Mutex
+	shells     = make(map[string]Interface)
+)
+
+func init() {
+	Register(&Raw{})
+	Register(&Bash{})
+}
+
+// Register a new shell interface.
+func Register(i Interface) {
```

Comment:
By keeping a register function (private or public) you can then add a file like "shell/cmd_windows.go" that registers itself when compiled.